cms/apps/media/forms.py

```diff
@@ -1,15 +1,49 @@
 import base64
 import os
 
+import magic
 from django.core.files.base import ContentFile
 from django import forms
 from io import BytesIO
 from PIL import Image
 
 from cms.apps.media.models import File
 
+CHECKED_FILETYPES = {
+    'image/jpeg',
+    'image/gif',
+    'image/png',
+    'image/tiff',
+}
 
-class ImageChangeForm(forms.ModelForm):
+
+# Takes a file and compares it's claimed MIME type to one calculated by python-magic,
+# but only if it's in a list of MIME types to check.
+def mime_check(file):
+    guessed_filetype = magic.from_buffer(file.read(1024), mime=True)
```

Comment:
I think this could result in corrupted files. It definitely will if whatever works with the file next does not remember to `seek(0)`. I'd recommend `file.seek(0)` immediately after this.